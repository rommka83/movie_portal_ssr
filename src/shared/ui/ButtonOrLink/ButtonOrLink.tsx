import React, { ButtonHTMLAttributes, MouseEvent } from 'react';
import styles from './buttonorlink.module.css';
import classNames from 'classnames';
import Link from 'next/link';

export const ButtonOrLink = React.memo<ButtonProps>(
  ({
    className,
    variant = 'primary',
    round,
    large,
    transparent,
    small,
    children,
    to,
    onClick,
    ...props
  }) => {
    const classes = classNames(styles.button, className, {
      [styles.primary]: variant === 'primary',
      [styles.secondary]: variant === 'secondary',
      [styles.third]: variant === 'third',
      [styles.round]: round,
      [styles.large]: large,
      [styles.transparent]: transparent,
      [styles.small]: small,
      [styles.disabled]: props.disabled,
    });

    if (to != null) {
      return (
        <Link className={classes} href={to} onClick={onClick}>
          {children}
        </Link>
      );
    }
    return (
      <button {...props} className={classes} onClick={onClick}>
        {children}
      </button>
    );
  },
);

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  variant?: 'primary' | 'secondary' | 'third';
  large?: boolean;
  transparent?: boolean;
  small?: boolean;
  round?: boolean;
  to?: string;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
};
