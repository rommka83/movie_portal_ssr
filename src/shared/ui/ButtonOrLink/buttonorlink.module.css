.button {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  font-weight: 500;
  font-size: 13px;
  line-height: 16px;
  border-width: 1px;
  border-style: solid;
  padding: 7px 11px;
  border-radius: var(--main-border-radius);
  color: #fff;
  transition-duration: 0.4s;
  cursor: pointer;
}

.button.primary {
  background-color: var(--pink);
  border-color: var(--pink);
}

.button.primary:hover {
  background-color: var(--pink-hover);
  border-color: var(--pink-hover);
}

.button.primary:active {
  background-color: var(--pink-active);
  border-color: var(--pink-active);
}

.button.secondary {
  background-color: var(--dark-gray);
  border-color: var(--dark-gray);
  padding: 9px 15px;
}

.button.secondary:hover {
  background-color: var(--dark-gray-hover);
  border-color: var(--dark-gray-hover);
}

.button.secondary:active {
  background-color: var(--dark-gray-active);
  border-color: var(--dark-gray-active);
}

.button.third {
  padding: 6px 16px;
  border-radius: var(--secondary-border-radius);
  background-color: var(--gray);
  color: var(--gray-light);
  border: none;
}

.button.third:hover {
  background-color: var(--gray-hover);
}

.button.third:active {
  background-color: var(--gray-active);
}

.button.transparent {
  background-color: transparent;
  border-color: var(--gray-32);
}

.button.transparent:hover {
  background-color: transparent;
  border-color: var(--gray-32);
}

.button.transparent:active {
  background-color: var(--gray-08);
  border-color: var(--gray-72);
}

.button.round {
  border-radius: 50%;
  height: 40px;
  width: 40px;
  padding: 0;
  border-width: 2px;
}

.button.small {
  padding: 3px 7px;
  border-radius: var(--third-border-radius);
}

.button.large {
  font-size: 15px;
  line-height: 20px;
}
