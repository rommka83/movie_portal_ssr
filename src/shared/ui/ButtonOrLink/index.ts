export { ButtonOrLink as default } from "./ButtonOrLink";
