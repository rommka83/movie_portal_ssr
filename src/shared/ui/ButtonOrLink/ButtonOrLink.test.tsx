import renderer from 'react-test-renderer';
import { ButtonOrLink } from './ButtonOrLink';

describe('проверка ButtonOrLink', () => {
  test('отрисовка ButtonOrLink primary, круглой', () => {
    const component = renderer.create(
      <ButtonOrLink variant='primary' round>
        Lorem ipsum dolor
      </ButtonOrLink>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('отрисовка ButtonOrLink secondary, большой', () => {
    const component = renderer.create(
      <ButtonOrLink variant='secondary' large>
        Lorem ipsum dolor
      </ButtonOrLink>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('отрисовка ButtonOrLink прозрачной, маленькой', () => {
    const component = renderer.create(
      <ButtonOrLink transparent small>
        Lorem ipsum dolor
      </ButtonOrLink>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
