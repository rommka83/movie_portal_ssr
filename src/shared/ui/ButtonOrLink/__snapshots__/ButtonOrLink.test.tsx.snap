// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка ButtonOrLink отрисовка ButtonOrLink primary, круглой 1`] = `
<button
  className="button primary round"
>
  Lorem ipsum dolor
</button>
`;

exports[`проверка ButtonOrLink отрисовка ButtonOrLink secondary, большой 1`] = `
<button
  className="button secondary large"
>
  Lorem ipsum dolor
</button>
`;

exports[`проверка ButtonOrLink отрисовка ButtonOrLink прозрачной, маленькой 1`] = `
<button
  className="button primary transparent small"
>
  Lorem ipsum dolor
</button>
`;
